make-mode: use sys.argv[0] instead of `sys.executable + "sphinx-build ..."`

### Problem
`make-mode` invokes `python sphinx-build arguments`. In our distribution we wrap all executable files and this is now broken because `sphinx-build` assumes it can just run `python sphinx-build`

#### Procedure to reproduce the problem
```
make html
```
e.g.

#### Error logs / results
```
$ make html
  File "/nix/store/r9p2wdbaz1ll67cqvr97zvhih7yvyhn2-python3.5-Sphinx-1.5.1/bin/sphinx-build", line 2
    export PATH=/nix/store/8qn95swyn4higc63c1rlaacvwkz6iaw7-python3-3.5.2/bin:/nix/store/r9p2wdbaz1ll67cqvr97zvhih7yvyhn2-python3.5-Sphinx-1.5.1/bin:/nix/store/rmhvpa9m94m4iniamnldy9y9l3kgx921-python3.5-docutils-0.13.1/bin:/nix/store/xrlbgkpz9ksvbhj11iwvv1844ryfy4i5-python3.5-setuptools-30.2.0/bin:/nix/store/f8ijxyzx7ilzxpbkpvydnqyxrxb13gx8-python3.5-Pygments-2.1.3/bin:/nix/store/i1zrk7ia2qac7mp3a39p9gmily04bxpv-python3.5-Babel-2.3.4/bin${PATH:+:}$PATH
              ^
SyntaxError: invalid syntax
make: *** [Makefile:20: html] Error 1
```
What you see here is the contents of our wrapper.

#### Expected results
If `sphinx-build` is an executable program, then it should be executed directly, instead of by calling Python on it.

We can [patch](https://github.com/NixOS/nixpkgs/commit/6640fe2e1e6b7f42fbf3a0c0648428c31c830c6e) it ourselves, but I think it is better if Sphinx wouldn't use Python to invoke `sphinx-build`. On most systems `sys.argv[0]` will return the full filename, so that should be sufficient.

### Environment info
- OS: Nix, NixOS
- Python version: Any
- Sphinx version: 1.5.1

Downstream issue https://github.com/NixOS/nixpkgs/issues/22501

--- a/CHANGES
+++ b/CHANGES
@@ -12,6 +12,7 @@ Features added
 
 * Support requests-2.0.0 (experimental) (refs: #3367)
 * (latex) PDF page margin dimensions may be customized (refs: #3387)
+* #3400: make-mode doesn't use subprocess on building docs
 
 Bugs fixed
 ----------

--- a/sphinx/make_mode.py
+++ b/sphinx/make_mode.py
@@ -19,9 +19,9 @@ from __future__ import print_function
 import os
 import sys
 from os import path
-from subprocess import call
 
 import sphinx
+from sphinx import cmdline
 from sphinx.util.console import bold, blue
 from sphinx.util.osutil import cd, rmtree
 
@@ -250,20 +250,12 @@ class Make(object):
         if doctreedir is None:
             doctreedir = self.builddir_join('doctrees')
 
-        orig_cmd = sys.argv[0]
-        if sys.platform == 'win32' and orig_cmd.endswith('.exe'):
-            # win32: 'sphinx-build.exe'
-            cmd = [orig_cmd]
-        elif sys.platform == 'win32' and os.path.splitext(orig_cmd)[1] == '':
-            # win32: 'sphinx-build'  without extension
-            cmd = [orig_cmd + '.exe']
-        else:
-            # win32: 'sphinx-build.py'
-            # linux, mac: 'sphinx-build' or 'sphinx-build.py'
-            cmd = [sys.executable, orig_cmd]
-
-        return call(cmd + ['-b', builder] + opts +
-                    ['-d', doctreedir, self.srcdir, self.builddir_join(builder)])
+        args = [sys.argv[0],
+                '-b', builder,
+                '-d', doctreedir,
+                self.srcdir,
+                self.builddir_join(builder)]
+        return cmdline.main(args + opts)
 
 
 def run_make_mode(args):

